rust/main/chains/hyperlane-sovereign/src/provider/rest_client.rs

```diff
@@ -411,6 +412,25 @@ impl SovereignRestClient {
         })
     }
 
+    async fn get_compensated_rollup_height(&self, lag: u64) -> ChainResult<u64> {
+        let (_, latest_batch) = self.get_latest_slot().await?;
+        let batch = self
+            .get_batch(
+                latest_batch
+                    .ok_or(ChainCommunicationError::CustomError(String::from(
+                        "latest batch was None",
+                    )))?
+                    .into(),
+            )
+            .await?;
+        match batch.rollup_height.checked_sub(lag) {
+            Some(compensated_height) => Ok(compensated_height),
+            None => Err(ChainCommunicationError::CustomError(String::from(
+                "lag was greater than rollup height",
+            ))),
+        }
+    }
```

Comment:
I wonder if we can write it much shorter. Something like:

```suggestion
    async fn get_compensated_rollup_height(&self, lag: u64) -> ChainResult<u64> {
        let latest_batch = self.get_latest_slot().await?.1.ok_or_else(|| {
            ChainCommunicationError::CustomError("latest batch was None".to_string())
        })?;

        let batch = self.get_batch(latest_batch.into()).await?;
    
        batch.rollup_height.checked_sub(lag).ok_or_else(|| {
            ChainCommunicationError::CustomError("lag was greater than rollup height".to_string())
        })
    }
```

Reply:
I am going to split the difference and take half of your suggestion. Thinking that using the `.1` is slightly less readable than a self-describing variable.